src/compression/huffman.rs

```diff
@@ -0,0 +1,114 @@
+use std::collections::{BTreeMap, BinaryHeap, HashMap};
+
+#[derive(Eq, PartialEq, Debug, Ord, PartialOrd)]
+pub enum HuffmanNode {
+    Internal {
+        frequency: usize,
+        left: Box<HuffmanNode>,
+        right: Box<HuffmanNode>,
+    },
+    Leaf {
+        character: char,
+        frequency: usize,
+    },
+}
+
+pub fn huffman_encode(text: &str) -> (String, BTreeMap<char, String>) {
+    let mut frequencies = HashMap::new();
+
+    for char in text.chars() {
+        let counter = frequencies.entry(char).or_insert(0);
+        *counter += 1;
+    }
```

Comment:
```suggestion
    let frequencies = text.chars().fold(HashMap::new(), |mut map, val| {
        *map.entry(val).or_insert(0) += 1;
        map
    });
```
I would suggest to move it into a separate function.